controllers/configmap_controller.go

```
@@ -183,7 +184,13 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context,
 	case servicescm.Name:
 		return ctrl.Result{}, r.reconcileServices(ctx, configMap)
 	case wiparser.InstanceConfigMap:
-		return ctrl.Result{}, r.reconcileNodes(ctx, configMap)
+		err := r.reconcileNodes(ctx, configMap)
+		var upgradeErr *UpgradeLimitExceededError
+		if errors.As(err, &upgradeErr) {
```

The `if` statement swallows the error and returns `ctrl.Result{}, nil`   will the event be reconciled (re-tried)?

---

it does get retried. THis is what the log looks like 


`{"level":"info","ts":"2025-07-25T16:55:29Z","logger":"controller.windowsmachine","msg":"instance requires upgrade","node":"ip-10-0-19-9.ec2.internal","version":"10.16.2-f61b7b2","expected version":"10.16.3-c406a61"}
{"level":"error","ts":"2025-07-25T16:55:29Z","msg":"Reconciler error","controller":"machine","controllerGroup":"machine.openshift.io","controllerKind":"Machine","Machine":{"name":"winworker-plhtr","namespace":"openshift-machine-api"},"namespace":"openshift-machine-api","name":"winworker-plhtr","reconcileID":"dd5256e0-1dcf-423e-bde7-de4619e60581","error":"unable to configure instance i-0a05e7d15e7a57e14: Cannot mark node ip-10-0-19-9.ec2.internal as upgrading. Current number of upgrading nodes is (1)","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/build/windows-machine-config-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/build/windows-machine-config-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2...
{"level":"info","ts":"2025-07-25T16:55:29Z","logger":"controller.windowsmachine","msg":"processing","windowsmachine":{"name":"winworker-tg6vr","namespace":"openshift-machine-api"},"address":"10.0.16.106"}
{"level":"info","ts":"2025-07-25T16:55:32Z","logger":"controller.windowsmachine","msg":"instance requires upgrade","node":"ip-10-0-16-106.ec2.internal","version":"10.16.2-f61b7b2","expected version":"10.16.3-c406a61"}
{"level":"error","ts":"2025-07-25T16:55:32Z","msg":"Reconciler error","controller":"machine","controllerGroup":"machine.openshift.io","controllerKind":"Machine","Machine":{"name":"winworker-tg6vr","namespace":"openshift-machine-api"},"namespace":"openshift-machine-api","name":"winworker-tg6vr","reconcileID":"0a996db4-c9ef-414f-86ee-a5ceadbe227a","error":"unable to configure instance i-03925c21cdc8f06c5: Cannot mark node ip-10-0-16-106.ec2.internal as upgrading. Current number of upgrading nodes is (1)","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/build/windows-machine-config-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/build/windows-machine-config-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.fun...
{"level":"info","ts":"2025-07-25T16:55:32Z","logger":"controller.windowsmachine","msg":"processing","windowsmachine":{"name":"winworker-4fqff","namespace":"openshift-machine-api"},"address":"10.0.28.83"}`

which repeats several times